test/test_euler.py

```diff
@@ -395,58 +395,33 @@ def test_facial_flux(ctx_factory, order, dim):
 
         discr = EagerDGDiscretization(actx, mesh, order=order)
 
-        mass_input = discr.zeros(actx)
-        energy_input = discr.zeros(actx)
+        mass_input = discr.zeros(actx) + 1.0
+        energy_input = discr.zeros(actx) + 2.5
         mom_input = flat_obj_array(
             [discr.zeros(actx) for i in range(discr.dim)]
         )
 
-        # This sets p = 1
-        mass_input[:] = 1.0
-        energy_input[:] = 2.5
-
         fields = flat_obj_array(
             mass_input, energy_input, mom_input
         )
 
         from mirgecom.euler import _facial_flux
 
         interior_face_flux = _facial_flux(
-            discr, q_tpair=interior_trace_pair(discr, fields)
-        )
-
-        err = np.max(
-            np.array(
-                [
-                    la.norm(interior_face_flux[i].get(), np.inf)
-                    for i in range(0, 2)
-                ]
-            )
-        )
+            discr, q_tpair=interior_trace_pair(discr, fields))
 
-        assert err < tolerance
+        from functools import partial
+        fnorm = partial(discr.norm, p=np.inf, dd="all_faces")
 
-        err1 = err
-        # mom flux max should be p = 1 + (any interp error)
-        maxmomerr = 0.0
-        for i in range(2, 2 + discr.dim):
-            err = np.max(
-                np.array(
-                    [
-                        la.norm(
-                            interior_face_flux[i].get(), np.inf
-                        )
-                    ]
-                )
-            )
-            err = np.abs(err - p0)
-            if err > maxmomerr:
-                maxmomerr = err
+        iff_split = split_conserved(dim, interior_face_flux)
+        assert fnorm(iff_split.mass) < tolerance
+        assert fnorm(iff_split.energy) < tolerance
 
-        assert err < tolerance
+        # FIXME (AK -> MC): explain why this is right
+        momerr = fnorm(iff_split.momentum) - p0
```

Comment:
This particularly seemed weird, given that the array seemed to contain values of 0, 1, and -1.

Reply:
(I.e. could you explain, maybe by adding a comment there?)

Reply:
Oh sure.  It is because the expected pressure 1.0 (by design).   And the flux diagonal is `[rho(v x v) + p]` since we have zero velocities it's just `p`.

Reply:
But what about the zero and the sign flip?

Reply:
The off-diagonals are zero. We get a {ndim}-vector for each dimension, the flux for the x-component of momentum (for example) is:
f_momx = < 1.0, 0 , 0> , then we return f_momx .dot. normal, which can have negative components.

Reply:
I dunno what happened there.  I hit the wrong button when trying to comment and it sent me into review mode.

Reply:
Thanks! Added your explanation in dfa591f.